pkg/reconciler/dependencybuild/scripts/maven-build.sh

```diff
@@ -27,7 +27,11 @@ EOF
 cp -r $(workspaces.source.path)/workspace $(workspaces.source.path)/source
 #we can't use array parameters directly here
 #we pass them in as goals
-mvn -B -e -s "$(workspaces.build-settings.path)/settings.xml" $@ "-DaltDeploymentRepository=local::file:$(workspaces.source.path)/artifacts" "org.apache.maven.plugins:maven-deploy-plugin:3.0.0-M2:deploy" | tee $(workspaces.source.path)/logs/maven.log
+mvn -B -e -s "$(workspaces.build-settings.path)/settings.xml" $@ "-DaltDeploymentRepository=local::file:$(workspaces.source.path)/artifacts" "org.apache.maven.plugins:maven-deploy-plugin:3.0.0-M2:deploy" | tee $(workspaces.source.path)/logs/maven-original.log
+
+rm -r $(workspaces.source.path)/artifacts
+
+unshare -n -Ufp -r -- mvn -B -e -s "$(workspaces.build-settings.path)/settings.xml" $@ "-DaltDeploymentRepository=local::file:$(workspaces.source.path)/artifacts" "org.apache.maven.plugins:maven-deploy-plugin:3.0.0-M2:deploy" | tee $(workspaces.source.path)/logs/maven.log
```

Comment:
@ralphbean @sbose78 @Michkov What do you guys think about this approach to Hermetic Java builds?

Basically we run the build normally, then delete the result and then re-run via 'unshare -n' which should prohibit network access. These results from the offline build are then used for deployment.

I am also wondering if we should be archiving all the dependencies used in the build, although creating an image with them for every build is not very space efficient.

Reply:
Do you have any data yet to suggest that this increases the failure rate of dependency rebuilds significantly?

Reply:
It does a bit, but they are mostly ant builds that are likely doing something they shouldn't anyway: 

Build with this enabled:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/redhat-appstudio_jvm-build-service/540/pull-ci-redhat-appstudio-jvm-build-service-main-jvm-build-service-in-repo-apicurio-e2e/1637756596568199168/artifacts/jvm-build-service-in-repo-apicurio-e2e/jvm-build-service-in-repo-apicurio-e2e/artifacts/jvm-build-service-report/index.html

Without:

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/redhat-appstudio_jvm-build-service/480/pull-ci-redhat-appstudio-jvm-build-service-main-jvm-build-service-in-repo-apicurio-e2e/1636172251504054272/artifacts/jvm-build-service-in-repo-apicurio-e2e/jvm-build-service-in-repo-apicurio-e2e/artifacts/jvm-build-service-report/index.html

Most of the failures seem to be related to a bug with the ant script that I will fix today and re-run. 

Micrometer seems to be the only genuine failure, it seems to still be attempting to hit the cache. Note that I can't use maven offline mode as the deploy plugin will fail in offline mode (even for local file deployments).

Reply:
Cool. 1 out of ~200 is not too bad. And, what does the path look like for a user who encounters a dependency like micrometer that genuinely fails? Is the only path to engage with upstream and agree on patches to build without going back out to cache? or, do you suspect there is some path via `jvm-build-data/build-deps/`?

In any case, I think I'm +1 to merge this and we can reap feedback from users. The increased confidence in provenance is worth the 1 failure in 200 tradeoff, in my opinion.